fedora_messaging/twisted/service.py

```diff
@@ -0,0 +1,97 @@
+# This file is part of fedora_messaging.
+# Copyright (C) 2018 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+from __future__ import absolute_import, unicode_literals
+
+import pika
+from twisted.application import service
+from twisted.application.internet import TCPClient, SSLClient
+from twisted.internet import ssl
+
+from .. import config
+from .._session import _configure_tls_parameters
+from .factory import FedoraMessagingFactory
+
+
+class FedoraMessagingService(service.MultiService):
+    """A Twisted service to connect to the Fedora Messaging broker."""
+
+    name = 'fedora-messaging'
+    factoryClass = FedoraMessagingFactory
+
+    def __init__(self, on_message, amqp_url=None, bindings=None):
+        """Initialize the service.
+
+        Args:
+            on_message (callable|None): Callback that will be passed each
+                incoming messages. If None, no message consuming is setup.
+            amqp_url (str): URL to use for the AMQP server.
```

Comment:
``__init__`` also accepts ``bindings``, looks like it's the same format documented in config